#DPATCHLEVEL=0
Author: Dann Frazier <dannf@hp.com>


2006-03-24  Dann Frazier  <dannf@hp.com>

	* main.c [HAVE_LIBREADLINE]: Include 'readline/readline.h'.


diff -urN ftp/main.c ftp/main.c
--- ftp/main.c	2002-04-29 14:40:30.000000000 -0600
+++ ftp/main.c	2005-07-17 04:21:01.282833004 -0600
@@ -55,6 +55,10 @@
 #include <unistd.h>
 #include <getopt.h>
 
+#if HAVE_LIBREADLINE
+#include <readline/readline.h>
+#endif
+
 /* Define macro to nothing so declarations in ftp_var.h become definitions. */
 #define FTP_EXTERN
 #include "ftp_var.h"


2006-03-24  Dann Frazier  <dannf@hp.com>

	* Makefile.am (noinst_HEADERS): Add 'utmp_init.h'.
	* utmp_init.h: New file.


--- headers/Makefile.am	2005-10-11 14:31:26.000000000 +0300
+++ headers/Makefile.am	2005-12-12 04:35:14.000000000 +0200
@@ -16,4 +16,4 @@
 # to the Free Software Foundation, Inc., 51 Franklin Street,
 # Fifth Floor, Boston, MA 02110-1301 USA.
 
-noinst_HEADERS = tftpsubs.h shishi_def.h
+noinst_HEADERS = tftpsubs.h shishi_def.h utmp_init.h
diff -urN headers/utmp_init.h headers/utmp_init.h
--- headers/utmp_init.h	1969-12-31 17:00:00.000000000 -0700
+++ headers/utmp_init.h	2005-07-17 04:04:55.086555778 -0600
@@ -0,0 +1,45 @@
+/* utmp_init.h -- 
+   Copyright (C) 2005 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+#ifndef UTMP_INIT_H_
+# define UTMP_INIT_H_
+
+# ifndef PARAMS
+#  if defined PROTOTYPES || (defined __STDC__ && __STDC__)
+#   define PARAMS(Args) Args
+#  else
+#   define PARAMS(Args) ()
+#  endif
+# endif
+
+# ifndef __attribute__
+#  if __GNUC__ < 2 || (__GNUC__ == 2 && __GNUC_MINOR__ < 8) || __STRICT_ANSI__
+#   define __attribute__(x)
+#  endif
+# endif
+
+# ifndef ATTRIBUTE_NORETURN
+#  define ATTRIBUTE_NORETURN __attribute__ ((__noreturn__))
+# endif
+
+/* utmp_init - update utmp and wtmp before login */
+void utmp_init(char *line, char *user, char *id);
+
+/* utmp_ptsid - generate utmp id for pseudo terminal */
+char *utmp_ptsid(char *line, char *tag);
+
+#endif /* !UTMP_INIT_H_ */


2006-03-24  Dann Frazier  <dannf@hp.com>

	* rlogind.c: Include 'utmp_init.h'. Add prototype for localhost.
	Move protypes for utmp_ptsid and utmp_init to 'utmp_init.h'.


diff -urN rlogind/rlogind.c rlogind/rlogind.c
--- rlogind/rlogind.c	2003-04-05 10:03:39.000000000 -0700
+++ rlogind/rlogind.c	2005-07-17 04:07:25.644171121 -0600
@@ -73,6 +73,8 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>		/* Needed for chmod() */
 
+#include "utmp_init.h"
+
 /*
   The TIOCPKT_* macros may not be implemented in the pty driver.
   Defining them here allows the program to be compiled.  */
@@ -130,6 +132,7 @@
 # define DEFPORT 513
 #endif
 extern int      __check_rhosts_file;
+extern char *localhost __P ((void));
 
 #ifndef SHISHI
 struct auth_data
@@ -634,9 +636,6 @@
 }
 
 #ifdef UTMPX
-char           *utmp_ptsid ();
-/*FIXME*/ void  utmp_init ();
-
 void
 setup_utmp (char *line)
 {


2006-03-24  Dann Frazier  <dannf@hp.com>

	* Makefile.am (INCLUDES): Add '-I$(top_srcdir)/headers'.
	* pty.c: Include 'utmp_init.h'.


--- telnetd/pty.c~	2002-04-07 08:31:37.000000000 -0600
+++ telnetd/pty.c	2005-07-17 06:51:11.121589821 -0600
@@ -18,6 +18,7 @@
    Fifth Floor, Boston, MA 02110-1301 USA. */

 #include "telnetd.h"
+#include "utmp_init.h"
 #include <sys/wait.h>

 #ifdef AUTHENTICATION
--- telnetd/Makefile.am	2005-10-11 14:29:14.000000000 +0300
+++ telnetd/Makefile.am	2005-12-12 04:39:13.000000000 +0200
@@ -30,7 +30,8 @@
 @PATHDEFS_MAKE@
 
 INCLUDES = $(PATHDEF_DEV) $(PATHDEF_TTY) $(PATHDEF_LOGIN) \
-	   -I$(top_srcdir) -I$(top_srcdir)/lib @INCAUTH@
+	   -I$(top_srcdir) -I$(top_srcdir)/headers -I$(top_srcdir)/lib \
+	   @INCAUTH@
 
 LIBTERMCAP = @LIBTERMCAP@
 LIBCRYPT = @LIBCRYPT@
